google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractStructReader.java

```diff
@@ -165,15 +177,20 @@ public boolean getBoolean(String columnName) {
 
   @Override
   public long getLong(int columnIndex) {
-    checkNonNullOfCodes(columnIndex, Arrays.asList(Code.ENUM, Code.INT64), columnIndex);
-    return getLongInternal(columnIndex);
+    checkNonNullOfCodes(
+        columnIndex, Arrays.asList(Code.ENUM, Code.PG_OID, Code.INT64), columnIndex);
+    return getColumnType(columnIndex).getCode() == Code.PG_OID
+        ? getPgOidInternal(columnIndex)
```

Comment:
I think that this can be simplified to just calling `getLongInternal`. The same also applies to the array methods below.

Reply:
Done.